src/firestore.rules.test.js

```diff
@@ -56,3 +61,38 @@ test('allows write to scholarships doc when signed in', () => {
     .set(newScholarship);
   return assertSucceeds(testDoc);
 });
+
+test('allows edit when you are author of scholarship', () => {
+  const testDoc = authedApp
+    .firestore()
+    .collection('scholarships')
+    .doc('KLJASDQW')
+    .set(newScholarship);
+
+  const updatedDoc = authedApp
+    .firestore()
+    .collection('scholarships')
+    .doc('KLJASDQW')
+    .set({ ...testDoc, name: 'updated name' });
+
+  return assertSucceeds(updatedDoc);
```

Comment:
`.set` returns a Promise you need to wait on to finish before you can update it.
```js
  return assertSucceeds(authedApp
    .firestore()
    .collection('scholarships')
    .doc('KLJASDQW')
    .set(newScholarship)
    .then((s) => s.set({ ...newScholarship, name: 'updated name' });
```

This might be what's causing the bug below too.